logstash-core/spec/api/lib/api/plugins_spec.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+require_relative "../../spec_helper"
+require "sinatra"
+require "app/modules/plugins"
+require "logstash/json"
+
+describe LogStash::Api::Plugins do
+
+  include Rack::Test::Methods
+
+  def app()
+    described_class
+  end
+
+  before(:all) do
+    get "/"
+  end
+
+  let(:payload) { JSON.parse(last_response.body) }
+
+  it "respond to plugins resource" do
+    expect(last_response).to be_ok
+  end
+
+  it "return valid json content type" do
+    expect(last_response.content_type).to eq("application/json")
+  end
+
+  it "return the expected schema" do
+    expect(payload.keys).to include("plugins", "total")
+  end
+
+  it "return a list of available plugins" do
+    payload["plugins"].each do |plugin|
+      expect(plugin).to be_available?
+    end
+  end
```

Comment:
we can do more in deep format check if you want, yes. But for now in all
test I tried to be a bit flexible as we still can change a lot of the
formats :-P

- purbon

On Fri, Mar 11, 2016 at 3:43 PM Pier-Hugues Pellerin <
notifications@github.com> wrote:

> In logstash-core/spec/api/lib/api/plugins_spec.rb
> <https://github.com/elastic/logstash/pull/4794#discussion_r55836906>:
>
> > +    expect(last_response).to be_ok
> > +  end
> > +
> > +  it "return valid json content type" do
> > +    expect(last_response.content_type).to eq("application/json")
> > +  end
> > +
> > +  it "return the expected schema" do
> > +    expect(payload.keys).to include("plugins", "total")
> > +  end
> > +
> > +  it "return a list of available plugins" do
> > +    payload["plugins"].each do |plugin|
> > +      expect(plugin).to be_available?
> > +    end
> > +  end
>
> @purbon <https://github.com/purbon> I checked the code before raising the
> issue, I am just saying I want to make sure we return the version key in
> the JSON response. I believe we need to have a test for that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4794/files#r55836906>.
>